src/datasets/load.py

```diff
@@ -761,6 +901,61 @@ def get_module(self) -> DatasetModule:
             use_auth_token=self.download_config.use_auth_token,
             timeout=100.0,
         )
+        config_kwargs, builder_kwargs = {}, {}
+
+        # this is moved here so that info from json will be rewritten by yaml is they both exist
+        download_config = self.download_config.copy()
+        if download_config.download_desc is None:
+            download_config.download_desc = "Downloading metadata"
+        try:
+            dataset_infos_path = cached_path(
+                hf_hub_url(self.name, config.DATASETDICT_INFOS_FILENAME, revision=self.revision),
+                download_config=download_config,
+            )
+            with open(dataset_infos_path, encoding="utf-8") as f:
+                dataset_infos: DatasetInfosDict = json.load(f)
+            dataset_info = dataset_infos.get(self.config_name, None)
+            if dataset_info:
+                builder_kwargs["info"] = DatasetInfo.from_dict(dataset_info)
+        except FileNotFoundError:
+            pass
+
+        download_config = self.download_config.copy()
+        if download_config.download_desc is None:
+            download_config.download_desc = "Downloading readme"
+        try:
+            dataset_readme_path = cached_path(
+                hf_hub_url(self.name, "README.md", revision=self.revision),
+                download_config=download_config,
+            )
+            dataset_metadata = DatasetMetadata.from_readme(Path(dataset_readme_path))
+        except FileNotFoundError:
+            dataset_metadata = DatasetMetadata()
+
+        metadata_configs_dict = (
+            MetadataConfigs.from_metadata(dataset_metadata) if dataset_metadata else MetadataConfigs()
+        )
+        config_kwargs = copy.deepcopy(metadata_configs_dict.get(self.config_name, {}))
+
+        if config_kwargs:
+            data_files = config_kwargs.pop("data_files", None)
+            data_dir = config_kwargs.pop("data_dir", None)
+
+            if data_files:
+                if self.data_files:
+                    logger.warning(
```

Comment:
no need to check this IMO - the `DatasetBuilder` is already able to overwrite an existing config with additional parameters and logs a warning